src/commissaire_http/bus/__init__.py

```diff
@@ -0,0 +1,196 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Bus related classes and functions.
+"""
+
+import logging
+import uuid
+
+from kombu import Connection, Exchange, Producer, Queue
+
+
+class Bus:
+    """
+    Connection to a bus.
+    """
+
+    def __init__(self, exchange_name, connection_url, qkwargs):
+        """
+        Initializes a new Bus instance.
+
+        :param exchange_name: Name of the topic exchange.
+        :type exchange_name: str
+        :param connection_url: Kombu connection url.
+        :type connection_url: str
+        :param qkwargs: One or more dicts keyword arguments for queue creation
+        :type qkwargs: list
+        """
+        self.logger = logging.getLogger('Bus')
+        self.logger.debug('Initializing bus connection')
+        self.connection = None
+        self._channel = None
+        self._exchange = None
+        self.exchange_name = exchange_name
+        self.connection_url = connection_url
+        self.qkwargs = qkwargs
+
+    @property
+    def init_kwargs(self):
+        """
+        Returns the initializing kwargs for this instance.
+
+        :returns: the initializing kwargs for this instance.
+        :rtype: dict
+        """
+        return {
+            'exchange_name': self.exchange_name,
+            'connection_url': self.connection_url,
+            'qkwargs': self.qkwargs,
+        }
+
+    def connect(self):
+        """
+        'Connects' to the bus.
+
+        :returns: The same instance.
+        :rtype: commissaire_http.bus.Bus
+        """
+        self.connection = Connection(self.connection_url)
+        self._channel = self.connection.channel()
+        self._exchange = Exchange(
+            self.exchange_name, type='topic').bind(self._channel)
+        self._exchange.declare()
+
+        # Create queues
+        self._queues = []
+        for kwargs in self.qkwargs:
+            queue = Queue(**kwargs)
+            queue.exchange = self._exchange
+            queue = queue.bind(self._channel)
+            self._queues.append(queue)
+            self.logger.debug('Created queue {}'.format(queue.as_dict()))
+
+        # Create producer for publishing on topics
+        self.producer = Producer(self._channel, self._exchange)
+        self.logger.debug('Bus connection finished')
+        return self
+
+    # TODO: Everything under this line needs to move to a common module
+    #       shared between service and http
+
+    @classmethod
+    def create_id(cls):  # pragma: no cover
+        """
+        Creates a new unique identifier.
+
+        :returns: A unique identification string.
+        :rtype: str
+        """
+        return str(uuid.uuid4())
```

Comment:
See my comment below about a mixin class in `commissaire`.  This method could also be moved there.